const_caller_location to use real Span instead of `DUMMY_SP`

cachebag opened the pull request:
Clarifying usage of DUMMY_SP

RalfJung commented:
What does the PR description have to do with the PR contents? The PR does not link anything to any issue, as far as I can tell.

jackh726 commented:
Hey @cachebag, thanks for the PR :)

I think we can land this, seems simple enough. Like @RalfJung said, we don't open issues just to link them in a PR, but that's okay.

Not sure how difficult it would be, but it'd be great if we could get this change covered by a test. Though, for this, I'm not sure it's necessary.

cachebag commented:
@RalfJung I see what you are saying, and it makes sense. Changed it.

@jackh726 Hi Jack, thanks for offering to keep this open for me. 

I can definitely see what it would take to test this. And after reading the guide fully this afternoon I can see why this PR raised eyebrows- I'll definitely make sure I follow proper contributing guidelines next time I come back here.

RalfJung commented:
That span is only relevant if an error occurs in this interpreter. I don't think it's possible for the interpreter to error in `const_caller_location_provider`. So I don't think a test is possible... and that also would mean it's likely not worth passing through the span if it'll anyway never be used. My first inclination would be to just replace the FIXME by a comment saying the span doesn't matter here since can be no errors.

@oli-obk knows this code better than me though, I may be missing something.

RalfJung commented:
Interpreters that run actual user code should have a span indicating the "root" of the evaluation. This is used in various places for error reporting.

But dummy interpreters like this one don't necessarily need a span.

cachebag committed:
const_caller_location: edit FIXME to explain use of `DUMMY_SP`

Co-authored-by: Ralf Jung <post@ralfj.de>

cachebag commented:
@RalfJung @oli-obk 

Thanks for the clarifications. I reverted the span plumbing, and just replaced the FIXME with the details you all mentioned. 

> also just bug! in case an error occurs

There's no need for another `bug!()` site right? I'm going to assume I should just leave it as it is, because I thought the only place errors can surface in `const_caller_location_provider` is after `intern_const_alloc_recursive`

RalfJung commented:
Looks good, thanks. :)

@bors r+ rollup

bors commented:
:pushpin: Commit c98b3b2da46146956b982e4ae1747c3614a461fd has been approved by `RalfJung`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=RalfJung c98b3b2da46146956b982e4ae1747c3614a461fd -->
<!-- homu: {"type":"Approved","sha":"c98b3b2da46146956b982e4ae1747c3614a461fd","approver":"RalfJung","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->
